app/services/user/squash.rb

```diff
@@ -0,0 +1,51 @@
+class User
+  class Squash < ApplicationService
+    def initialize(params = {})
+      @params       = params
+      @squshed_user = params.fetch(:squshed_user, nil)
+      @into_user    = params.fetch(:into_user, nil)
+    end
+
+    def call
+      return unless schema.call(params).success?
+      return if squshed_user.id == into_user.id
+
+      fix_has_many_dependencies!
+      destroy_squashed_user!
```

Comment:
How about to wrap these calls in a transaction, so if one fails, we don't get a partly-squashed user?
```ruby
ActiveRecord::Base.transaction do
  fix_has_many_dependencies!
  destroy_squashed_user!
end
```